File: AutoSTP/backend/app/schemas/document.py
"""Схемы API документов."""
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


class DocumentVersionOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    version_string: str
    created_at: datetime
    snapshot: dict[str, Any] | None


class DocumentListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    title: str
    document_type: str
    status: str
    created_at: datetime
    updated_at: datetime
    original_filename: str | None = None


class DocumentDetail(BaseModel):
    id: str
    title: str
    document_type: str
    status: str
    metadata: dict[str, Any] | None = None
    created_at: datetime
    updated_at: datetime
    current_version: DocumentVersionOut | None
    versions_count: int


class FormatRequest(BaseModel):
    template_id: str = Field(..., description="UUID шаблона для форматирования")


class DocumentUpdate(BaseModel):
    title: str | None = Field(default=None, max_length=512)
    snapshot: dict[str, Any]


class DocumentListResponse(BaseModel):
    items: list[DocumentListItem]
    total: int
    skip: int
    limit: int

File: AutoSTP/frontend/src/types/document.ts
export type DocumentType = 'ku' | 'di'
export type DocumentStatus = 'uploaded' | 'inpr' | 'formatted'

export interface DocumentSnapshot {
  plain_text: string
  nodes: unknown[]
}

export interface DocumentVersion {
  version_string: string
  snapshot: DocumentSnapshot
  created_at: string
}

export interface DocumentMetadata {
  stored_file_format: string
  upload_extension: string
  stored_filename: string
  storage_dir: string
}

export interface DocumentListItem {
  id: string
  title: string
  document_type: DocumentType
  status: DocumentStatus
  created_at: string
  updated_at: string
  original_filename?: string
}

export interface DocumentDetail extends DocumentListItem {
  current_version: DocumentVersion
  versions_count: number
  metadata: DocumentMetadata
}

export interface DocumentListResponse {
  items: DocumentListItem[]
  total: number
  skip: number
  limit: number
}

export interface SectionOut {
  id: string
  role: string
  title: string
  level: number
  order_number: number
  text_preview: string
  char_count: number
}

export interface SegmentResponse {
  document_id: string
  template_id: string
  sections: SectionOut[]
  total_sections: number
  unmatched_chars: number
}

export interface HintsResponse {
  section_id: string
  hints: string[]
}
